src/lib/eth-calls.ts

```diff
@@ -88,9 +94,25 @@ async function lockController(
     contractAddress,
     this.signer
   );
-  const tx = await contract.lockController(caller, tableId);
+
+  const opts = await getOverrides(this.signer);
+  const tx = await contract.lockController(caller, tableId, opts);
 
   return await tx.wait();
 }
 
+async function getOverrides(signer: Signer): Promise<Overrides> {
```

Comment:
Nice, I quite like how you did this, pretty non-invasive, and easy to rip out if price estimates get back on track downstream. Just to be certain, it looks like if signer _doesn't_ have a provider this will just skip along and return empty opts (which is good), correct?

Reply:
yep, but btw, looks like https://github.com/tablelandnetwork/js-tableland/blob/main/src/lib/check-network.ts#L15 enforces provider is always set by the time we get to the contract calls.

Reply:
Ah yeh, I wasn't paying attention. This is in `eth-calls`... so we do actually need a provider here always anyway 👍 .